src/pqm/pqm.py

```diff
@@ -216,36 +567,58 @@ def pqm_pvalue(
         For best results, we suggest using a large number of re-tessellations,
         though this is our recommendation in any case.
 
+
     Returns
     -------
     float or list
         pvalue(s). Null hypothesis that both samples are drawn from the same
         distribution.
     """
+    # Check the device and convert to the respective type (Numpy or Torch) and call their respective _pqm_test function
+
+    if device.type == 'cpu':
```

Comment:
This won't work if someone enters `device="cpu"` since str doesn't have the attribute `type`